src/main/java/com/checkout/ApiClientImpl.java

```diff
@@ -179,23 +177,38 @@ private Response errorCheck(final Response response) {
         return response;
     }
 
-    private <T> T deserialize(final Response response, final Class<T> responseType) {
-        final T result = serializer.fromJson(response.getBody(), responseType);
+    private <T extends HttpMetadata> T deserialize(final Response response, final Class<T> responseType) {
+        T result = serializer.fromJson(response.getBody(), responseType);
+        //Unfortunate but GSON returns null if body is null, we need to instantiate the class type parameter for adding metadata
+        if (result == null) {
```

Comment:
What if we do a `new EmptyResponse()` in this situation?

Reply:
I don't this that could work, because certain methods expects an specific response, so that might cause some cast steps checks